### PR TITLE
ifpack2 : add option to use Zoltan2 for testing with belos

### PR DESCRIPTION
**Pull request by iyamazaki**
@trilinos/ifpack2 

## Motivation

Add an option to use Zoltan2 for testing ifpack2 with belos

### CODE REVIEW

File: packages/ifpack2/test/belos/build_problem.hpp [hunk 1]
```diff
@@ -222,6 +232,61 @@ build_problem (Teuchos::ParameterList& test_params,
...
+    // Redistribute Sol
+    RCP<TMV> zoltan_x;
+    Zoltan2::XpetraMultiVectorAdapter<TMV> adapterSol(rcpFromRef (*x));
+    adapterSol.applyPartitioningSolution (*x, zoltan_x, problem.getSolution());
+    // Set it as Sol
+    x = zoltan_x;
+#else
+    TEUCHOS_TEST_FOR_EXCEPTION(
+      useZoltan2, std::invalid_argument,
+      "Both Xpetra and Zoltan2 are neeeded to usee Zoltan2.");
```
Comment (jhux2) [inline packages/ifpack2/test/belos/build_problem.hpp:286]:
```suggestion
      "Both Xpetra and Zoltan2 are needed to use Zoltan2.");
```

**Review by jhux2 — CHANGES_REQUESTED** [review]
@iyamazaki I think you also need to add `Zoltan2` to https://github.com/trilinos/Trilinos/blob/401873f5da440ebe7a6e8c4569f3dd6e2fbec2c6/packages/ifpack2/cmake/Dependencies.cmake#L4.

File: packages/ifpack2/test/belos/build_problem.hpp (hunk 1 above)
Comment (cgcgcg) [reply packages/ifpack2/test/belos/build_problem.hpp:286]: Not sure:
https://github.com/trilinos/Trilinos/blob/401873f5da440ebe7a6e8c4569f3dd6e2fbec2c6/packages/ifpack2/cmake/Dependencies.cmake#L2
and
https://github.com/trilinos/Trilinos/blob/401873f5da440ebe7a6e8c4569f3dd6e2fbec2c6/packages/ifpack2/cmake/Ifpack2_config.h.in#L94-L98

Comment (iyamazaki) [reply packages/ifpack2/test/belos/build_problem.hpp:286]: Thank you. I could use Zoltan2 without having it in the list. But I locally made the fixes, so I'll push once the auto-tester finish.

File: packages/ifpack2/test/belos/build_problem.hpp [hunk 2]
```diff
@@ -283,7 +283,7 @@ build_problem (Teuchos::ParameterList& test_params,
 #else
     TEUCHOS_TEST_FOR_EXCEPTION(
       useZoltan2, std::invalid_argument,
-      "Both Xpetra and Zoltan2 are neeeded to usee Zoltan2.");
+      "Both Xpetra and Zoltan2 are needed to usee Zoltan2.");
```
Comment (jhux2) [inline packages/ifpack2/test/belos/build_problem.hpp:286]:
```suggestion
      "Both Xpetra and Zoltan2 are needed to use Zoltan2.");
```

sorry, one more typo

**Review by jhux2 — APPROVED** [review]
lgtm

**Pull request merged**